Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -509,13 +509,18 @@ std::map<std::string, FieldValue> DecodeObject(pb_istream_t* stream) {
 
 void Serializer::EncodeFieldValue(const FieldValue& field_value,
                                   std::vector<uint8_t>* out_bytes) {
-  // TODO(rsgowman): how large should the output buffer be? Do some
-  // investigation to see if we can get nanopb to tell us how much space it's
-  // going to need. (Hint: use a sizing stream, i.e. PB_OSTREAM_SIZING)
-  uint8_t buf[1024];
-  pb_ostream_t stream = pb_ostream_from_buffer(buf, sizeof(buf));
+  pb_ostream_t stream = {
+      /*callback=*/[](pb_ostream_t* stream, const pb_byte_t* buf,
+                      size_t count) -> bool {
+        auto* out_bytes = static_cast<std::vector<uint8_t>*>(stream->state);
```

Comment:
Optional nit: consider making this a reference, since there's no check for nullptr anyway.

Reply:
I kept it a pointer to stay consistent with the definition of out_bytes in the outer scope. (Left as is for now... but I don't feel strongly about it.)

Reply:
I generally prefer the most restricted form that works (reference is more restricted because it's non-resettable and can't be null), but I'm totally fine with you leaving it as is.